### ISSUE TITLE
Rapidly completing job hangs instead of finishing

### ISSUE THREAD

**Issue by LiterallyUniqueLogin** [snakemake/snakemake#1323]
**Snakemake version**
6.12.3

**Describe the bug**
My job completes in under a second, waits 5 minutes, and then is cancelled. Instead, the snakemake process should notice that it's done and wrap up like normal.

**Logs**
Notice the start and end times. Also notice that snakemake sees that the job is done, but somehow cancels it anyways.
```
Building DAG of jobs...
Updating job finemap_signal_strengths.
Updating job export_phenotype_summaries.
Updating job manhattan_finemap.
Using shell: /usr/bin/bash
Provided cores: 1 (use --cores to define parallelism)
Rules claiming more threads will be scaled down.
Select jobs to execute...

[Mon Jan 10 08:45:05 2022]
rule export_phenotype_summaries:
    input: ...
    output: export/.exported
    jobid: 0
    wildcards: phenotype=albumin
    resources: mem_mb=1000, disk_mb=1000, tmpdir=/tmp, threads=1, mem_gb=5, attempt=0, time=00:05:00

... link some files ...

touch export/.exported
[Mon Jan 10 08:45:05 2022]
Finished job 0.
1 of 1 steps (100%) done
slurmstepd: error: *** JOB 8297525 ON exp-1-15 CANCELLED AT 2022-01-10T08:45:29 DUE TO TIME LIMIT ***
```

**Minimal example**
```
rule export_phenotype_summaries:
    input:
        ...
    output:
        # owns all symlinks in this directory
        complete='export/.exported'
    resources:
        time='00:05:00',
        mem_gb=5 # got OOM error for this job, don't know why
    run: 
        # delete all symlinks in the out directory before running
        out_dir = output.complete[:output.complete.rfind('/')] # strip the file off of the directory
        out_links = [path for path in pathlib.PosixPath(out_dir).iterdir() if path.is_symlink()]
        for path in out_links:
            path.unlink()

        for file_ in input:
            link_file(
                file_,
                'export/'
            )

        shell(f"touch {output.complete}")
```

**Comment by iamh2o** [comment]
I have observed similar behavior across several recent version of snakemake.  This is, for me, intertwined with the bug described [here](https://github.com/snakemake/snakemake/issues/1244).  I mitigate by adding a sleep 60 globally to the end of all rules.

**Comment by johanneskoester** [comment]
For me, this works fine with latest main branch (will be released as 7.0 soon). To keep the management simple: I will close this now, and watch out for you reopening it in case I am wrong. Thanks a lot for reporting!
